test/sync/src/commonMain/kotlin/io/realm/entities/sync/SyncObjectWithAllTypes.kt

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.entities.sync
+
+import io.realm.ObjectId
+import io.realm.RealmInstant
+import io.realm.RealmList
+import io.realm.RealmObject
+import io.realm.annotations.PrimaryKey
+import io.realm.realmListOf
+import io.realm.test.util.TypeDescriptor
+import kotlin.random.Random
+
+private typealias FieldDataFactory = (SyncObjectWithAllTypes) -> Unit
+private typealias FieldValidator = (SyncObjectWithAllTypes) -> Unit
+
+class SyncObjectWithAllTypes : RealmObject {
+    @PrimaryKey
+    var _id: String = "id-${Random.nextLong()}"
+
+    // Non-nullable types
+    var stringField: String = "hello world"
+    var byteField: Byte = 0
+    var charField: Char = 0.toChar()
+    var shortField: Short = 0
+    var intField: Int = 0
+    var longField: Long = 0
+    var booleanField: Boolean = true
+    var doubleField: Double = 0.0
+    var floatField: Float = 0.0.toFloat()
+    var realmInstantField: RealmInstant = RealmInstant.MIN
+    var objectIdField: ObjectId = ObjectId.create()
+    var objectField: SyncObjectWithAllTypes? = null
+
+    // Nullable types
```

Comment:
should we include `RealmList` of nullable as well for each type?

Reply:
They are not supported by Sync: https://github.com/realm/realm-core/issues/5545... I'll add a comment here about it.